### ISSUE TITLE
Installing a prerelease package using --version fails to resolve prerelease dependencies

### ISSUE THREAD

**Issue by jberezanski** [chocolatey/choco#1575]
### What You Are Seeing?

(I'm fairly certain I've seen it discussed somewhere earlier, but I couldn't find any existing issue, so here it is.)

Given 

1) a package `main-package` with a prerelease version, for example `2.0.0-pre`
2) a package `sub-package`, also with a prerelease version, for example `1.2.3-pre`
3) main-package 2.0.0-rc depends on sub-package 1.2.3-rc

invoking

````
choco install main-package --version 2.0.0-rc
````

results in failure, with choco complaining "Unable to resolve dependency 'sub-package (= 1.2.3-rc)'.".

On the other hand, if choco is invoked explicitly with `--pre`:

````
choco install main-package --version 2.0.0-rc --pre
````

installation succeeds

### What is Expected?

When told to install a specific package version, when that version is a prerelease, choco should be able to resolve prerelease dependencies, without requiring the user to specify `--pre` explicitly.

### How Did You Get This To Happen? (Steps to Reproduce)

````
choco install dotnetcore-windowshosting --version "2.1.0.20180510-rc1" -y
````

(source: [Disqus comment](http://disq.us/p/1shz0kg))

### Output Log

https://gist.github.com/jberezanski/8541f715850e387089936b44757083ca

**Comment by bcurran3** [comment]
IMHO: choco should install the dependencies specified and not attempt any "readjusting" of them.

 i.e. if a dependency is specified as a release version, choco installs the release version. If the dependency is specified as a pre-release version, choco installs the pre-release version. So a pre-release package with pre-release dependencies should install those pre-release dependencies because that's what was specified. IMHO It doesn't make sense for choco to install a pre-release package and then also search for pre-released dependencies if they exist. Makes more sense to define the pre-release dependencies in the pre-release package until all are released and the issue becomes moot.

**Comment by martincostello** [comment]
The package where the issue was found did define a prerelease dependency of its own, but it failed to install without `--pre` being specified when installing the prerelease pacakage that declared it as a dependency.

**Comment by bcurran3** [comment]
Thanks. WITH THAT SAID... it seems choco should install whatever version it's damn well told to do -pre or not! :)

**Comment by ckolumbus** [comment]
Any progress on this one (any direction)?

**Comment by gravity00** [comment]
Any news about this? Still having the same issues to manage

**Comment by corbob** [comment]
I have done some testing of this issue, and from what I can tell it is now resolved in 2.3.0. It also looks like it was not resolved in 2.2.2. As such, I'll close this issue and tag it to the 2.3.0 milestone. If for some reason my testing is incorrect, we can reopen this.
